### PR TITLE
Fix push notifications android bug

### LINKED ISSUE #8550: Never ending loading indicator is shown on white screen if open app from push notification
### Description

[comment]: # (Feature or Bug? i.e Type: Bug)

On Android: white screen with never ending loading is shown if open app from push notification. Full logcat is also attached but the cause might be this error:
```
07-10 11:27:24.792 14445 14488 E ReactNativeJS: ERROR [status-im.utils.fx:31] - Merging fx with common-keys:  #{:db :status-im.ui.screens.navigation/navigate-to}
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: console.error: "ERROR [status-im.utils.fx:31] - Merging fx with common-keys:  #{:db :status-im.ui.screens.navigation/navigate-to}", stack:
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: hca@11:86988
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: a@11:89969
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: e@11:1958201
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: l@11:1958046
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: <unknown>@11:702880
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: <unknown>@11:706641
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: Ib@11:145660
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: n@11:27054
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: pe@11:67422
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: MO@11:705938
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: vBb@11:724761
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: <unknown>@11:725275
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: md@11:47268
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: ob@11:49822
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: h@11:26829
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: h@11:68164
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: o@11:725207
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: wrc@11:2052700
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: <unknown>@11:2054547
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: <unknown>@11:725290
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: md@11:47268
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: ob@11:49822
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: h@11:26829
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: h@11:68164
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: o@11:725207
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: a@11:2054242
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: <unknown>@11:2054116
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: <unknown>@11:725290
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: md@11:47268
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: ob@11:49822
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: h@11:26829
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: h@11:68164
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: o@11:725207
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: <unknown>@11:2242120
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: <unknown>@11:664157
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: cwb@11:656014
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: gwb@11:657164
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: jwb@11:659247
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: iwb@11:657487
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: <unknown>@11:658477
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: QN@11:659114
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: n@11:657551
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: y@80:657
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: C@80:1021
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: callImmediates@80:3216
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: callImmediates@-1
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: value@19:2873
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: <unknown>@19:1264
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: value@19:2565
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: value@19:1234
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: value@-1
07-10 11:27:24.893 14445 14489 E unknown:ReactNative: value@-1
```

Screenshot_20190710-112955

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)

#### Expected behavior

[comment]: # (Describe what you expected to happen.)
Chat with a message from contact is opened
#### Actual behavior

[comment]: # (Describe what actually happened.)
White screen with loading indicator is shown for at least 5 minutes
### Reproduction

[comment]: # (Describe how we can replicate the bug step by step.)

- two users add each other as contacts an send some text msg in 1:1 chat
- user1 on Android closes the app
- user2 sends a text msg in 1;1 chat
- user1 gets push notification and taps on it.  Check if user1 ends up in the 1:1 chat with user2

### Additional Information

[comment]: # (Please do your best to fill this out.)
- Status version: nightly from July 10
- Operating System: Android 9 (real pixel XL)

#### Logs

[comment]: # (if on Android please replicate bug whilst running adb logcat)

[not2.log](https://github.com/status-im/status-react/files/3376806/not2.log)

### PR DESCRIPTION
**Pull request by vitvly**
Fixes #8550 

Kudos to @yenda for explaining how `fx/merge` and `mergable-keys` work.

Seems that this is what happened: `open-login-callback` navigates to `:home`, but then, within the same `fx/merge` call, we might also get another `:navigate-to` fx if there are push notifications that can open a chat window. So I've tried to check if that is the case, and if so, drop `:home` navigation.

iOS workaround for 1ms delayed navigation is also removed as probably this should not be necessary anymore.

### CODE REVIEW

**Review by flexsurfer — APPROVED** [review]

**Comment by asemiankevich** [comment]
@siphiuel i noticed that notifications pop-up appears here without tapping the appropriate button (enable notification settings). It does not happen in nightly. Could you please take a look?
https://drive.google.com/open?id=1H_kc7D33SlZBYo5sLAKR9PdNXlmvxPXH

**Comment by vitvly** [comment]
@asemiankevich fixed it, sorry, dumb mistake

**Review by yenda — APPROVED** [review]

File: src/status_im/init/core.cljs [hunk 1]
```diff
@@ -204,22 +204,14 @@
...
-            (when-not (or (:intro-wizard db) platform/ios?)
-              (navigation/navigate-to-cofx :home nil))
-            (notifications/process-stored-event address stored-pns)
-            (when platform/desktop?
-              (chat-model/update-dock-badge-label))))
+  (let [cofx (fx/merge cofx (notifications/process-stored-event cofx address stored-pns))]
+    (fx/merge cofx
+              (when-not (:intro-wizard db)
+                {:notifications/request-notifications-permissions nil})
+              (when-not (or (:intro-wizard db) (:status-im.ui.screens.navigation/navigate-to cofx))
```
Comment (yenda) [inline src/status_im/init/core.cljs]: ::navigation/navigate-to and require status-im.ui.screens.navigation :as navigation

Comment (vitvly) [reply src/status_im/init/core.cljs]: Fixed, thanks!

**Comment by asemiankevich** [comment]
@siphiuel Vitaly, i noticed the following:

image


So i can't login when i am redirected to new login screen from push notification on android, if the app was closed before.

Steps:
1. add device 1 (ios) and device2 (android) to contacts
2. exchange some messages in 1x1 chat
3. close the app on android
4. send message from ios to android
5. wait for PN on android and tap it -> proceed to login

Please check the video: https://drive.google.com/open?id=1QLe14YY4aYiaet7rmK-1rOrU0fOYuYBm

**Comment by oskarth** [comment]
What's the state of this PR @siphiuel?

**Comment by yenda** [comment]
I think we should close this one and go back to it after status-go account persistence is merged

**Pull request closed without merging**